app/customer.py

```diff
@@ -0,0 +1,72 @@
+import datetime
+import math
+from typing import Dict, Any, List
+
+from app.car import Car
+
+
+class Customer:
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.name = data["name"]
+        self.product_cart = data["product_cart"]
+        self.location = data["location"]
+        self.money = data["money"]
+        self.car = Car(data["car"])
+
+    def calculate_trip_cost(self, shop: "Shop") -> float:
+        distance_to_shop = self.calculate_distance(shop.location)
+        fuel_consumption_per_km = self.car.fuel_consumption / 100
+        fuel_cost_to_shop = (
+            distance_to_shop * fuel_consumption_per_km
+        ) * Car.FUEL_PRICE
+        product_cost = sum(
+            [
+                self.product_cart[product] * shop.products[product]
+                for product in self.product_cart
+            ]

```

Comment:
Do not use list comprehension inside of `sum`, use generator expression instead.